address.go

```diff
@@ -175,24 +173,24 @@ type UTXOIDChainID interface {
 	FromOutputID(id OutputID) ChainID
 }
 
-func newAddress(addressType AddressType) (address Address, err error) {
+func newAddress(addressType AddressType) Address {
 	switch addressType {
 	case AddressEd25519:
-		return &Ed25519Address{}, nil
+		return &Ed25519Address{}
 	case AddressAccount:
-		return &AccountAddress{}, nil
+		return &AccountAddress{}
 	case AddressNFT:
-		return &NFTAddress{}, nil
+		return &NFTAddress{}
 	case AddressAnchor:
-		return &AnchorAddress{}, nil
+		return &AnchorAddress{}
 	case AddressImplicitAccountCreation:
-		return &ImplicitAccountCreationAddress{}, nil
+		return &ImplicitAccountCreationAddress{}
 	case AddressMulti:
-		return &MultiAddress{}, nil
+		return &MultiAddress{}
 	case AddressRestricted:
-		return &RestrictedAddress{}, nil
+		return &RestrictedAddress{}
 	default:
-		return nil, ierrors.Wrapf(ErrUnknownAddrType, "type %d", addressType)
+		panic(fmt.Sprintf("unknown address type %d", addressType))
```

Comment:
With that change, the `ParseBech32` which takes user input could panic on unknown address types.
I don't think this is a good idea.